app/Repositories/Summit/DoctrineSummitRegistrationPromoCodeRepository.php

```diff
@@ -272,59 +272,165 @@ public function getBySummit
         Order $order   = null
     )
     {
-        $query = $this->getEntityManager()
-                    ->createQueryBuilder()
-                    ->select("pc")
-                    ->from($this->getBaseEntity(), "pc")
-                    ->leftJoin(SummitRegistrationDiscountCode::class, 'dc', 'WITH', 'pc.id = dc.id')
-                    ->leftJoin(MemberSummitRegistrationDiscountCode::class, 'mdc', 'WITH', 'pc.id = mdc.id')
-                    ->leftJoin(SpeakerSummitRegistrationDiscountCode::class, 'spkdc', 'WITH', 'pc.id = spkdc.id')
-                    ->leftJoin(SpeakersRegistrationDiscountCode::class, 'spksdc', 'WITH', 'pc.id = spksdc.id')
-                    ->leftJoin(SponsorSummitRegistrationDiscountCode::class, 'spdc', 'WITH', 'pc.id = spdc.id')
-                    ->leftJoin(MemberSummitRegistrationPromoCode::class, 'mpc', 'WITH', 'pc.id = mpc.id')
-                    ->leftJoin(SponsorSummitRegistrationPromoCode::class, 'spc', 'WITH', 'mpc.id = spc.id')
-                    ->leftJoin(SpeakerSummitRegistrationPromoCode::class, 'spkpc', 'WITH', 'spkpc.id = pc.id')
-                    ->leftJoin(SpeakersSummitRegistrationPromoCode::class, 'spkspc', 'WITH', 'spkspc.id = pc.id')
-                    ->leftJoin(PrePaidSummitRegistrationPromoCode::class, 'pppc', 'WITH', 'pppc.id = pc.id')
-                    ->leftJoin(PrePaidSummitRegistrationDiscountCode::class, 'ppdc', 'WITH', 'ppdc.id = pc.id')
-                    ->leftJoin('pc.summit', 's');
-
-        $query = $this->applyExtraJoins($query, $filter);
-
-        $query = $query->where("s.id = :summit_id");
+        $extra_filters = sprintf(" WHERE pc.SummitID = %s", $summit->getId());
+        $extra_orders = '';
+        $bindings = [];
 
-        $query->setParameter("summit_id", $summit->getId());
-
-        if(!is_null($filter)){
-            $filter->apply2Query($query, $this->getFilterMappings($filter));
+        if ($filter instanceof Filter) {
+            $where_conditions = $filter->toRawSQL([
+                'code'          => 'pc.Code',
+                'description'   => 'pc.Description',
+                'class_name'    => 'pc.ClassName',
```

Comment:
class name is not working correctly
bc param sent on filter is for example class_name==SUMMIT_PROMO_CODE and u need to translate to
SummitRegistrationPromoCode::class ( check original implementation here https://github.com/OpenStackweb/summit-api/blob/570b8d9ce02913a3cbd789d15c3d23eaf7b63eb9/app/Repositories/Summit/DoctrineSummitRegistrationPromoCodeRepository.php#L140)